### ISSUE TITLE
Turning off Infinite approval UX confusion

### ISSUE THREAD

**Issue by alexwhte** [connext/monorepo#3118]
I might not understand the UX, please help me if I misunderstand 
https://testnet.bridge.connext.network/

After you allow infinite approval and transfer once, future transfers are always already approved. 

However if you transfer again and turn this setting off, im not sure it actually turns it off for you? It didnt look to work for me in sending a transfer

![image](https://user-images.githubusercontent.com/37959170/213371334-63861d64-5ff6-4376-b958-889ea3842489.png)

If user needs to confirm this part in their wallet, then 
I'm not sure if a slider button and waiting for the user to submit makes the most sense. 

Instead we might have a 'Turn off Infinite approval" option that triggers this for user to accept in their wallet --- separate from submitting a transfer

We should also try to retain this as a user setting for that wallet rather than automatically turning it back on in future transfers

**Comment by pjenviri** [comment]
The infinite approval is used when calling `sdk.nxtpSdkBase.approveIfNeeded`. The UI never knows whether it has been approved or how much it was approved in the past. Therefore, the infinite approval setting on the UI works only for passing a Boolean to the SDK's xcall function.

**Comment by pjenviri** [comment]
Please suggest how we should handle/update this.

**Comment by rhlsthrm** [comment]
We could check the amount approved and not display this slider at all if the approval amount is sufficient. But @pjenviri is correct, the UI won't really know if it has been approved in the past and I'm not sure a "user setting" needs to be there for this since it can be approved outside of this UI.

**Comment by pjenviri** [comment]
@rhlsthrm , now we have `approveIfNeeded` on the SDK. Can you add an additional method: `isApproveNeeded` (or other terms), that returns a boolean value? 

We can add it on the UI side, but I think it'd be easier to manage if we add it to the SDK.

**Comment by rhlsthrm** [comment]
`approveIfNeeded` already returns undefined if there's enough approval right? That seems like it should be enough.

**Comment by pjenviri** [comment]
@rhlsthrm yes. but understand that it will also call [populateTransaction](https://github.com/connext/monorepo/blob/main/packages/agents/sdk/src/sdkShared.ts#L158), which may not be used for purpose. Is that okay to do so?

**Comment by pjenviri** [comment]
@rhlsthrm @alexwhte done. Please try it and let us know if you see any issues.

**Comment by rhlsthrm** [comment]
> @rhlsthrm yes. but understand that it will also call [populateTransaction](https://github.com/connext/monorepo/blob/main/packages/agents/sdk/src/sdkShared.ts#L158), which may not be used for purpose. Is that okay to do so?

Yes this is fine! All this does is create transaction data.

**Comment by alexwhte** [comment]
@pjenviri  @rhlsthrm 
I wanted to turn infinite approval off but now it doesnt even show me it's on? what am i missing
![image](https://user-images.githubusercontent.com/37959170/216373545-5d4689f6-7ee3-419a-984c-04ba35063415.png)

**Comment by alexwhte** [comment]
or youre saying user should go and turn off any approvals in Explorers

**Comment by rhlsthrm** [comment]
Yeah, if you infinite approve it won't just turn it off, I don't think our UI was set up to remove approvals. Anyways other apps do this too, you have to revoke approvals elsewhere.

**Comment by alexwhte** [comment]
Instead of hiding should we grey it out? We can show a tooltip that says you need to revoke approval outside. So user is aware it's on?

**Comment by rhlsthrm** [comment]
> Instead of hiding should we grey it out? We can show a tooltip that says you need to revoke approval outside. So user is aware it's on?

I like this! We don't need to say "it's on" as in infinite approval, but we can just say "Approval sufficient, revoke using other tools" or something.

**Comment by alexwhte** [comment]
@rhlsthrm What do you mean by sufficient? Im trying to dumb it down for someone who doesnt understand well what infinite approval is

@pjenviri  When we come up with a tooltip can you please unhide this but have it greyed out where a user has infinite approval?

**Comment by rhlsthrm** [comment]
The tooltip can say: "Approval sufficient. If you need to, please revoke using other tools."

**Comment by rhlsthrm** [comment]
QAed.
